zscript/Weapons/LASERDEFS

```diff
@@ -0,0 +1,55 @@
+// Lasers begin here!
+class LaserSpriteInitalizer : Actor
+{
+	States
+	{
+		Sprites: 
+			X027 A -1;
+			X027 B -1;
+			X027 C -1;
+			X027 D -1;
+			X027 E -1;
+			X027 F -1;
+			X027 G -1;
+			X027 H -1;
+			X027 I -1;
+			X027 J -1;
+			X027 K -1;
+	}
+}
+
+class FreezeRifleLaser : MWLaser
+{
+	Default
+	{
+		DamageFunction 30;
+		DamageType "Ice";
+		speed 320;
+		MWLaser.frontOffset -100;
+		MWLaser.range 9000;
+		MWLaser.laserSprite "X027";
+		MWLaser.laserSize 0.25;
+		MWLaser.speedFactor 2.0;
+		Radius 4;
+		Height 8;
+		Decal "FreezerBurnSmall";
+		+NODAMAGETHRUST;
+		+ICEDAMAGE;
+	}
+	states
+	{
+		death:
+			TNT1 A 0 A_SpawnItemEx ("DetectFloorIce",0,0,1,0,0,0,0,SXF_NOCHECKPOSITION,0);
+			TNT1 A 0 A_SpawnItemEx ("DetectCeilIce",0,0,1,0,0,0,0,SXF_NOCHECKPOSITION,0);
+			BXPL A 0 Bright A_StopSound(CHAN_BODY);
+			BXPL A 0 Bright A_ChangeFlag("ICEDAMAGE", 1);
```

Comment:
This A_ChangeFlag isn't needed anymore and the ICEDAMAGE flag is deprecated. DamageType "Ice" does the trick and should be used instead. A_ChangeFlag is deprecated, use {bFlagName = true/false;} instead. There are other deprecated functions but as of right now I won't point them out until I find it.

Reply:
yea i kinda just copy pasted stuff over from the projectile originally just to get it workin, ill probably get those deprecated features out another day